index.js

```diff
@@ -616,6 +616,9 @@ function speedTest(options) {
   nextServer();
 
   function gotServers(err, servers) {
+    if (servers.settings == undefined) {
```

Comment:
There's 2 things that concern me here:
- If there was an error already, then `err` is set (and probably, `servers` is `undefined`). The error would be overwritten if neither `err` nor `servers` were `undefined`.
- If there was an error, `servers` will probably be `undefined`, leading to an error being thrown when attempting to access `servers.settings`

I think it's best to use the following:
```
if (!err && (!servers.settings || !servers.settings.servers || !servers.settings.servers.length)) {
```

This will make sure nothing happens if we already had an error, and thoroughly check the `servers` variable.